File: backend/schemas/inventario.py
"""
Schemas Pydantic para el módulo de inventario.
Items: catálogo por empresa — empresa_id explícito en Create.
Asignaciones: empresa_id heredado del ítem al crear (no lo provee el usuario).
"""
from datetime import date, datetime
from typing import List, Optional
from uuid import UUID

from pydantic import BaseModel


# ── Ítems de inventario ────────────────────────────────────────────────────────

class ItemCreate(BaseModel):
    empresa_id: UUID
    nombre: str
    tipo: str
    descripcion: Optional[str] = None
    numero_serie: Optional[str] = None
    fecha_alta: Optional[date] = None
    costo: Optional[float] = None
    notas: Optional[str] = None


class ItemUpdate(BaseModel):
    nombre: Optional[str] = None
    tipo: Optional[str] = None
    descripcion: Optional[str] = None
    numero_serie: Optional[str] = None
    costo: Optional[float] = None
    notas: Optional[str] = None
    estado: Optional[str] = None


class ItemResponse(BaseModel):
    id: str
    empresa_id: str
    empresa_nombre: Optional[str] = None
    nombre: str
    tipo: str
    descripcion: Optional[str] = None
    numero_serie: Optional[str] = None
    estado: str
    fecha_alta: date
    costo: Optional[float] = None
    notas: Optional[str] = None
    asignado_a: Optional[str] = None  # nombre del empleado que lo tiene actualmente
    created_at: datetime


class ItemListResponse(BaseModel):
    items: List[ItemResponse]
    total: int


# ── Asignaciones ───────────────────────────────────────────────────────────────

class AsignacionCreate(BaseModel):
    item_id: UUID
    empleado_id: UUID


class DevolucionRequest(BaseModel):
    estado_devolucion: str  # "ok" | "con_daño"
    notas: Optional[str] = None


class AsignacionResponse(BaseModel):
    id: str
    empresa_id: str
    empresa_nombre: Optional[str] = None
    item_id: str
    item_nombre: Optional[str] = None
    item_tipo: Optional[str] = None
    item_numero_serie: Optional[str] = None
    empleado_id: str
    empleado_nombre: Optional[str] = None
    fecha_asignacion: date
    fecha_devolucion: Optional[date] = None
    estado_devolucion: Optional[str] = None
    notas: Optional[str] = None
    created_at: datetime


class AsignacionListResponse(BaseModel):
    items: List[AsignacionResponse]
    total: int

File: frontend/types/inventario.ts
export type EstadoItem = "disponible" | "asignado" | "en_reparacion" | "baja"
export type EstadoDevolucion = "ok" | "con_daño"

export interface InventarioItem {
  id: string
  empresa_id: string
  empresa_nombre: string | null
  nombre: string
  tipo: string
  descripcion: string | null
  numero_serie: string | null
  estado: EstadoItem
  fecha_alta: string   // "YYYY-MM-DD"
  costo: number | null
  notas: string | null
  asignado_a: string | null  // nombre del empleado que lo tiene actualmente
  created_at: string
}

export interface InventarioItemCreate {
  empresa_id: string
  nombre: string
  tipo: string
  descripcion?: string
  numero_serie?: string
  costo?: number
  notas?: string
}

export interface InventarioItemUpdate {
  nombre?: string
  tipo?: string
  descripcion?: string
  numero_serie?: string
  costo?: number
  notas?: string
}

export interface ItemListResponse {
  items: InventarioItem[]
  total: number
}

export interface Asignacion {
  id: string
  empresa_id: string
  empresa_nombre: string | null
  item_id: string
  item_nombre: string | null
  item_tipo: string | null
  item_numero_serie: string | null
  empleado_id: string
  empleado_nombre: string | null
  fecha_asignacion: string   // "YYYY-MM-DD"
  fecha_devolucion: string | null
  estado_devolucion: EstadoDevolucion | null
  notas: string | null
  created_at: string
}

export interface AsignacionCreate {
  item_id: string
  empleado_id: string
}

export interface DevolucionRequest {
  estado_devolucion: EstadoDevolucion
  notas?: string
}

export interface AsignacionListResponse {
  items: Asignacion[]
  total: number
}
